pkg/provider/azure_loadbalancer.go

```diff
@@ -1239,30 +1236,32 @@ func (az *Cloud) isFrontendIPChanged(
 	lbFrontendIPConfigName string,
 	subnet *network.Subnet,
 ) (bool, error) {
-	isServiceOwnsFrontendIP, isPrimaryService, err := az.serviceOwnsFrontendIP(config, service)
-	if err != nil {
-		return false, err
-	}
+	isServiceOwnsFrontendIP, isPrimaryService, fipIPVersion := az.serviceOwnsFrontendIP(config, service)
 	if isServiceOwnsFrontendIP && isPrimaryService && !strings.EqualFold(pointer.StringDeref(config.Name, ""), lbFrontendIPConfigName) {
 		return true, nil
 	}
 	if !strings.EqualFold(pointer.StringDeref(config.Name, ""), lbFrontendIPConfigName) {
 		return false, nil
 	}
-	isInternal := requiresInternalLoadBalancer(service)
-	isIPv6, err := az.isFIPIPv6(service, &config, isInternal)
-	if err != nil {
-		return false, err
+	pipRG := az.getPublicIPAddressResourceGroup(service)
+	var isIPv6 bool
+	var err error
+	if fipIPVersion != "" {
+		isIPv6 = fipIPVersion == network.IPv6
+	} else {
+		if isIPv6, err = az.isFIPIPv6(service, pipRG, &config); err != nil {
+			return false, err
+		}
 	}
 	loadBalancerIP := getServiceLoadBalancerIP(service, isIPv6)
+	isInternal := requiresInternalLoadBalancer(service)
 	if isInternal {
 		// Judge subnet
 		subnetName := getInternalSubnet(service)
 		if subnetName != nil {
-			if err := az.fillSubnet(subnet, *subnetName); err != nil {
-				return false, err
+			if subnet == nil {
+				return false, fmt.Errorf("isFrontendIPChanged: Unexpected nil subnet")
```

Comment:
nit: can we include subnet name in the error msg?

Reply:
Let's do it in the next PR.